test/terra/states/test_aer_state.py

```diff
@@ -0,0 +1,465 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019, 2020, 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Integration Tests for Parameterized Qobj execution, testing qasm_simulator,
+statevector_simulator, and expectation value snapshots.
+"""
+
+import unittest
+from math import pi
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit, Gate
+from qiskit.quantum_info.random import random_unitary
+from qiskit.quantum_info.states.random import random_statevector
+from qiskit.providers.aer import AerSimulator
+
+from test.terra import common
+from qiskit.providers.aer.aererror import AerError
+
+class TestAerState(common.QiskitAerTestCase):
+    """AerState tests"""
+
+    def test_load_library(self):
+        """Test load library."""
+        from qiskit.providers.aer.backends.controller_wrappers import AerStateWrapper
+        state_wrapper = AerStateWrapper()
+
+    def test_generate_aer_state(self):
+        """Test generation of AerState"""
+        from qiskit.providers.aer.quantum_info.states.aer_state import AerState
+        state = AerState()
+
+    def test_move_from_aer_state(self):
+        """Test move of aer state to python"""
+        from qiskit.providers.aer.quantum_info.states.aer_state import AerState
+        state = AerState()
+        state.allocate_qubits(4)
+        state.initialize()
+        sv = state.move_to_ndarray()
+        state.close()
+
+    def test_error_reuse_aer_state(self):
+        """Test reuse AerState after move of aer state to python"""
+        from qiskit.providers.aer.quantum_info.states.aer_state import AerState
+        state = AerState()
+        state.allocate_qubits(4)
+        state.initialize()
+        sv = state.move_to_ndarray()
+
+        try:
+            state.allocate_qubits(4)
+            self.fail("do not reach here")
+        except:
+            pass
```

Comment:
This should be something like:

```suggestion
        with self.assertRaises(Exception):
            state.allocate_qubits(4)
```

and you should make the exception specific to what `AerState` actually raises as that is part of the API.